src/globus_sdk/experimental/auth_requirements_error/_validators.py

```diff
@@ -0,0 +1,65 @@
+from __future__ import annotations
+
+import typing as t
+
+from ._serializable import Serializable
+
+S = t.TypeVar("S", bound=Serializable)
+
+
+class ValidationError(ValueError):
+    pass
+
+
+def str_(name: str, value: t.Any) -> str:
+    if not isinstance(value, str):
+        raise ValidationError(f"'{name}' must be a string")
+    return value
+
+
+def opt_str(name: str, value: t.Any) -> str | None:
+    if value is None:
+        return None
+    if not isinstance(value, str):
+        raise ValidationError(f"'{name}' must be a string")
```

Comment:
As above, I recommend ending on `ValidationError`.

In addition, should this note that the field could be `None` or `null` as well?

Reply:
Similar questions for the other `opt_*` validators as well.